### PR TITLE
[Artwork] Fix season-all posters and manage context menu item

### LINKED ISSUE #18634: *All seasons" section poster not displaying
### Describe the bug
Here is a clear and concise description of what the problem is:

In the "*All seasons" section of TV Shows a folder icon is displayed instead of a poster. 

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

A poster should display.

## Actual Behavior

A folder icon displays

## Possible Fix

Goddam it Jim I'm a doctor not a programmer!

### To Reproduce

1. N/A
2.
3.

### Debuglog

The debuglog can be found here:

### Screenshots 
Here are some links or screenshots to help explain the problem:

Untitled

https://i.ibb.co/QDz01h9/Untitled.jpg

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ ] Linux
 - [ ] OSX
 - [tick ] Windows
 - [ ] Windows UWP

 - Operating system version/name:
 - Kodi version:

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

### PR DESCRIPTION
**Pull request by enen92**
## Description
This fixes a feature that has been broken since Kodi Krypton (included). Users can have `season-all-*`  artwork (e.g. season-all-poster.png) in the root folder of a show. When using metadata.local scrapper this artwork is taken into account and added to the database, being later displayed when selecting "All". Furthermore, users can use the "manage" context menu item to modify the artwork of the "All seasons" listitem.

When the "All Seasons" listitem is created, the list of items containing all the seasons of the show that is present on the mediawindow **also includes the top level (..) fileitem**. The infotag can come from any of the listitems in the list...except the first. This was the one being used :) 

## Motivation and Context
Fixes https://github.com/xbmc/xbmc/issues/18634

## How Has This Been Tested?
Library of shows made of strm files with nfos scraped via tinymediamanager. Added season-all-poster.jpg to one of the shows

## Screenshots (if appropriate):

**Before:**

Before

**Now:**

Now

Nowmanage

Note: I used a random picture from another show just to test the functionality.

## Types of change

- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

**Comment by KarellenX** [comment]
I scanned 5 tv shows from local artwork and nfo files. I added a random season-all-poster.jpg poster to each show. 
- The season-all-poster was correctly scanned and cached for each show.
- I was able to navigate the Context Menu>Manage>Choose Art and change the season-all poster to another poster.

I scraped 4 tv shows from online. The Big Bang Theory being one of them.
https://fanart.tv/series/80379/the-big-bang-theory/
- I used XML version of TheMovieDB by @olympia but this did not scrape season-all poster
- I used Python version of TheMovieDB by @pkscout  but this did not scrape season-all poster
- I then tried TVDB XML version and this did scrape season-all poster. 
-- I was also able to navigate Context Menu>Manage>Choose Art and change poster. Unfortunately this points to tv show Posters and not Season posters. Not sure if this is a scraper issue or Kodi core.

@enen92 Let me know if the final issue is scraper or core related.

**Comment by enen92** [comment]
Thanks for the tests @KarellenX. I need to try it myself and you provided enough info for me to do it. I've only tested metadata.local with nfo files so I am unsure what the other code path does 👍

**Comment by DaveTBlake** [comment]
Also is `...` always visible and always at the top? I'm sure there are options for that on on music nav screens

**Comment by KarellenX** [comment]
The ellipses can be disabled with setting 1.1
https://kodi.wiki/view/Settings/Media/General

**Comment by enen92** [comment]
The ".." parent directory can be hidden using the setting @KarellenX refers. However, when the setting is enabled and the directory is added it is always inserted at the top (first item) - c.f. https://github.com/xbmc/xbmc/blob/5230b683323ca58c62459a371c1306a6cb4d4644/xbmc/windows/GUIMediaWindow.cpp#L787 of the list. When it reaches this point, the list of seasons is already populated.

Regarding the findings @KarellenX provides for the scrapped media items I was able to reproduce using the tvdb scrapper and "the big bang theory" show:

- The scrapper correctly inserts a poster for all seasons and assigns it the -1 season id. The poster is a generic poster, however tvdb doesn't seem to provide "all season" specific posters.

- When using the manage item in the "All seasons" directory item, a list of generic posters for the show is in fact displayed. This seems to be intentional. Kodi iterates over all the artwork the show and if the type (e.g. poster) matches , it is added to the list. This occurs if this is the generic show item or the all seasons (-1) item. Cf: https://github.com/xbmc/xbmc/blob/5230b683323ca58c62459a371c1306a6cb4d4644/xbmc/utils/ScraperUrl.cpp#L112

In summary, I think everything is fine with the PR and it can safely be merged. Furthermore I tested all the other "season-all.*" combination images and all are working as expected:

allseasons

**Review by KarellenX — APPROVED** [review]
Works as expected. 

Thanks for looking at it @enen92

**Review by DaveTBlake — APPROVED** [review]
Thanks @enen92
